libpod/container_config.go

```diff
@@ -382,6 +382,9 @@ type ContainerMiscConfig struct {
 	// IsInfra is a bool indicating whether this container is an infra container used for
 	// sharing kernel namespaces in a pod
 	IsInfra bool `json:"pause"`
+	// IsService is a bool indicating whether this container is a service container used for
+	// tracking the life cycle of K8s service.
+	IsService bool `json:"isService"`
```

Comment:
It feels like containers with this attribute need some sort of deletion protection, to ensure they aren't removed while pods referring to them are still present. Alternatively, we could remove it from relevant pods when doing a `RemoveContainer`?

Reply:
Something I will tackle in a follow-up PR.  I already have a prototype that marries service-containers with systemd.

I think the service container should work for the pods as an infra does for a single pod.  Without --force, it a removal will error out.  Probably need to do the same for stop.